core/rpc/src/author/tests.rs

```diff
@@ -44,17 +44,39 @@ fn uxt(sender: AccountKeyring, nonce: u64) -> Extrinsic {
 	tx.into_signed_tx()
 }
 
+struct TestSetup {
+	pub runtime: runtime::Runtime,
+	pub client: Arc<Client<Backend>>,
+	pub keystore: BareCryptoStorePtr,
+	pub pool: Arc<Pool<FullChainApi<Client<Backend>, Block>>>,
+}
+
+impl TestSetup {
+	fn author(&self) -> Author<Backend, Executor, FullChainApi<Client<Backend>, Block>, RuntimeApi> {
+		Author {
+			client: self.client.clone(),
+			pool: self.pool.clone(),
+			subscriptions: Subscriptions::new(Arc::new(self.runtime.executor())),
+			keystore: self.keystore.clone(),
+		}
+	}
+}
+
+fn setup() -> TestSetup {
```

Comment:
Maybe `TestSetup::default` or `TestSetup::new` to keep everything in a single place?